pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -356,6 +368,16 @@ func (s *Service) getOrCreateMonitor(openStackCluster *infrav1.OpenStackCluster,
 		Timeout:        5,
 	}
 	monitor, err = s.loadbalancerClient.CreateMonitor(monitorCreateOpts)
+	// Skip creating monitor if it is not supported by Octavia provider
+	var errUnexpectedResponseCode gophercloud.ErrUnexpectedResponseCode
+	if errors.As(err, &errUnexpectedResponseCode) {
+		if errUnexpectedResponseCode.Actual == http.StatusNotImplemented {
+			record.Warnf(openStackCluster, "SkippedCreateMonitor", "Health Monitor is not implemented with the current provider.")
+			record.Eventf(openStackCluster, "SkippedCreateMonitor", "Health Monitor is not created as it's not implemented with the current provider.")
+			return nil
+		}
+	}
```

Comment:
Looks pretty good. I can suggest to move this to utils/errors/errors.go as `IsNotImplementedError()` to skip some logic here.

Reply:
Fixed